codemod_yaml/box/yaml/sequence.py

```diff
@@ -31,16 +31,41 @@ def append(self, other: Any) -> None:
         seq_item = PyBlockSequenceItem(other, yaml_style=self._yaml_style)
 
         self.stream.edit(self, seq_item, append=True)
-        self._items[self._count] = other
-        self._count += 1
+        self._items.append(other)
+
+    @overload
+    def __getitem__(self, index: slice) -> list[Union[BoxedYaml, BoxedPy]]: ...
+    @overload
+    def __getitem__(self, index: int) -> Union[BoxedYaml, BoxedPy]: ...
+    def __getitem__(self, index):  # type: ignore[no-untyped-def]
+        if isinstance(index, slice):
+            start, stop, step = index.indices(len(self._items))
+            return [self[i] for i in range(start, stop, step)]
 
-    def __getitem__(self, index: int) -> Union[BoxedYaml, BoxedPy]:
-        self._ensure(index)
         value = self._items[index].value  # note: lazy property
         assert isinstance(value, (BoxedYaml, BoxedPy))
         return value
 
-    def __setitem__(self, index: int, other: Any) -> None:
+    @overload
+    def __setitem__(self, index: slice, other: Sequence[Any]) -> None: ...
```

Comment:
I had no idea `slice` is a type.

Reply:
Yeah, although the syntax itself is annoyingly specialcased, you can only use it in `<rvalue>[<slice>]`